images/microOS/config.xml

```diff
@@ -603,12 +603,8 @@
         <package name="lvm2"/>
         <package name="chrony"/>
         <package name="tuned"/>
-        <package name="python3-pip"/>
-        <package name="python3-aiofiles"/>
-        <package name="python3-requests"/>
         <package name="python3-rados"/>
-        <package name="python3-websockets"/>
```

Comment:
Why those dependencies are dropped?

Reply:
because we're now relying on a virtualenv to run stuff from. These dependencies are being installed in the virtualenv.

Reply:
This is a wrong approach, if the venv is prepared on the host. The host system is not necessary the same as the guest one.

Reply:
Can you be more specific?

Reply:
but I guess it is done in a right way

Reply:
> Can you be more specific?

for example, host system is Leap, guest system is Tumbleweed, there two different system package dependencies as well as python3 version itself, if you bundle venv on host system and try run it on guest one, you'll fail.

Reply:
As we have failed for the Leap case in this particular PR.

Reply:
Same true for guys who are trying this out on Ubuntu or RH hosts.

Reply:
They're embedded in the tarball now inside a python venv

Reply:
Why do we need to embed dependencies into tarball? Why not to use pip to install?

Reply:
I want to ultimately make a package of aquarium (rpm, deb, etc.), so i'm making a venv inside the tarball and using pip to install in that venv when the tarball is built. The implication of this, which I apparently hadn't realised until yesterday when the leap CI failed, is that the the host you build the tarball on, basically has to be the same OS the tarball is deployed on. That will (or should) eventually work for the case of providing an aquarium package for any distro, but right now breaks when using any non-tumbleweed distro to build the MicroOS image.

Reply:
@tserong maybe use kiwi for that?